### PR TITLE
Implement unwind hooking for aarch64 to allow backtracing through stalked code

### PR DESCRIPTION
**Pull request by s1341**

### CODE REVIEW

File: gum/backend-arm64/gumstalker-arm64.c [hunk 1]
```diff
@@ -808,6 +834,218 @@ gum_stalker_init (GumStalker * self)
...
+  ip = _Unwind_GetIP (context);
+
+  real_address = gum_metal_hash_table_lookup (ctx->excluded_calls, (gconstpointer) ip);
+  if (real_address == NULL)
+  {
+    real_address = gum_invocation_stack_translate (gum_interceptor_get_current_stack(),
+        (gpointer) ip);
+    if (real_address == NULL)
+    {
+      real_address = gum_metal_hash_table_lookup (ctx->mappings, (gconstpointer) ip);
```
Comment (oleavr) [inline gum/backend-arm64/gumstalker-arm64.c:1039]: The values in this one are `GumExecBlock *` values, so this looks like it needs some work.

File: gum/backend-arm64/gumstalker-arm64.c [hunk 2]
```diff
@@ -462,6 +472,22 @@ struct _GumBackpatch
   };
 };
 
+extern _Unwind_Reason_Code __gxx_personality_v0 (int version,
```
Comment (oleavr) [inline gum/backend-arm64/gumstalker-arm64.c:475]: I guess all of this should be behind:

```C
#if defined (HAVE_LINUX) && !defined (HAVE_ANDROID)
```

Like in the x86 backend?

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks! 🙌

File: gum/backend-arm64/gumstalker-arm64.c [hunk 3]
```diff
@@ -4577,6 +4821,9 @@ gum_exec_block_write_call_invoke_code (GumExecBlock * block,
     else
       gum_arm64_writer_put_blr_reg (cw, target->reg);
 
+    gum_metal_hash_table_insert (ctx->excluded_calls, gc->instruction->end,
+        cw->code);
```
[the remark this answers is not in the document]
Comment (s1341) [reply gum/backend-arm64/gumstalker-arm64.c:4825]: It seems to work as is...

File: gum/backend-arm64/gumstalker-arm64.c (hunk 1 above)
Comment (s1341) [reply gum/backend-arm64/gumstalker-arm64.c:1039]: What do I need to do?

File: gum/backend-arm64/gumstalker-arm64.c (hunk 2 above)
Comment (s1341) [reply gum/backend-arm64/gumstalker-arm64.c:475]: Nope... I tested on aarch64 android.

File: gum/backend-arm64/gumstalker-arm64.c (hunk 1 above)
Comment (oleavr) [reply gum/backend-arm64/gumstalker-arm64.c:1027]: I don't know, but will figure it out.

**Pull request merged**